### ISSUE TITLE
Support for Android

### ISSUE THREAD

**Issue by arunshan** [leecade/react-native-swiper#53]
I am planning to use this module for a project and would like to know if this works on Android. All the components used in this repo does not have 'IOS' suffix to it, so I believe it should be fine. :)

**Comment by GasperKolenc** [comment]
It doesn't really work on android. There is no paging, just continuous scroll. Paging buttons also don't work.

**Comment by lucasmreis** [comment]
+1 for Android support

**Comment by niftylettuce** [comment]
:+1: For Android!

**Comment by jforaker** [comment]
@niftylettuce do you have a fork with working swipe?  (already borrowed your navbar fork :stuck_out_tongue_winking_eye:)

**Comment by GasperKolenc** [comment]
For anyone looking for paging functionality, check out this module: https://github.com/race604/react-native-viewpager

**Comment by cgarvis** [comment]
I didn't like the datasource of reac-native-viewpager and it does look like the auther is going to be working on this repo soon, so I built https://github.com/fixt/react-native-page-swiper

**Comment by sritampatnaik** [comment]
@cgarvis Does your's support Android ?

**Comment by cgarvis** [comment]
@sritampatnaik yes.

**Comment by niftylettuce** [comment]
@jforaker unfortunately no :laughing:

**Comment by niftylettuce** [comment]
Perhaps we could borrow the Modal/Portal concept from my package here (which ensures the swiper would be shown above everything else).  https://github.com/niftylettuce/react-native-loading-spinner-overlay

**Comment by raymanyoung** [comment]
In fact the issue of Android support is that ViewPagerAndroid is used instead of scrollView so the adding corresponding event will make it work

**Comment by bsinc** [comment]
+1 would really appreciate this!

**Comment by sharq1** [comment]
+1

**Comment by acamenhas** [comment]
+1 !!!

**Comment by samdturner** [comment]
This looks like a terrific project, and we are seriously considering using it in production for our iOS app.  Is releasing a version with Android support going to happen soon - like in the next couple of months? (we obviously would use the same component for iOS and Android).  Your hard work is really appreciated!

**Comment by darrenchiu** [comment]
this module is awesome, please do add android support! Love this module so much

**Comment by narek11** [comment]
android support

**Comment by anonrig** [comment]
+1 for android support

**Comment by shireen** [comment]
+1

**Comment by sdduursma** [comment]
+1

**Comment by yohanceli** [comment]
+1

**Comment by oliversisson** [comment]
Update? @leecade

**Comment by oliversisson** [comment]
So it works fine for me in Android so long as I add
```
containerStyle={{width: windowDimensions.width}}
```

**Comment by jaspreetsingh2701** [comment]
+1

**Comment by yarrumretep** [comment]
+1

**Comment by nickitatkach** [comment]
+1

**Comment by plattyp** [comment]
Per @oliversisson comment, I'm able to get it to work fine in both iOS and Android by doing the following with the `containerStyle` prop and width:

```js
...
import Dimensions from 'Dimensions';

class Scene extends Component {
  getWidth() {
    return Dimensions.get('window').width;
  }

  render() {
    const width = this.getWidth();

    return (
       <View style={styles.outerContainer}>
          <Swiper
            showsButtons={false}
            showsPagination={false}
            containerStyle={{width}}
          >
            {this.renderCards()}
          </Swiper>
       </View>
    );
}
```

**Comment by joshkmartinez** [comment]
+1 for android support! I really need this!

**Comment by RaonyMarcondes** [comment]
+1 Android Support

**Comment by sejas** [comment]
+1 Android Support

**Comment by sejas** [comment]
yeah, @plattyp  is right. It is already working in Android (check the [Roadmap](https://github.com/leecade/react-native-swiper/blob/master/ROADMAP.md))

So we should close this issue.

```
import Dimensions from 'Dimensions';

class Slider extends Component {
  render() {
    const width = Dimensions.get('window').width
    return (
       <View style={styles.outerContainer}>
          <Swiper
            showsButtons={false}
            showsPagination={false}
            containerStyle={{width}}
          >
            {this.renderCards()}
          </Swiper>
       </View>
    );
}
```

**Comment by zacharymorel** [comment]
The android fix doesn't work for me even when I add containerStyle={{width}} as a Swiper prop.

**Comment by joshkmartinez** [comment]
@zacharymorel Same for me. Its cut off at the top and bottom on android, absolutely fine on ios...

**Comment by joshkmartinez** [comment]
@zacharymorel Have you had a fix for this yet?

**Comment by zacharymorel** [comment]
@joshkmartinez I haven't yet, I've decided to attempt to build a carousel component myself.

**Comment by azwar** [comment]
@zacharymorel make sure you are passing Style object.

Here is my detail example:

`        <Swiper
            showsButtons={false}
            showsPagination={false}
            containerStyle={swiperStyle}
          >
            {this.renderCards()}
       </Swiper>`

`const styles = StyleSheet.create({
    swiperStyle: {
        width: Platform.OS === 'ios' ? '100%' : Dimensions.get('window').width
    }
})`

**Comment by anasidrissi** [comment]
worked just fine for me 
just watch out for any containing views make sure to assign the appropriate style

**Comment by RossKinsella** [comment]
`automaticallyAdjustContentInsets` sorted out android not rendering for me...

RN: 0.54.4
react-native-swiper: 1.5.13

**Comment by brunokiafuka** [comment]
> `automaticallyAdjustContentInsets` sorted out android not rendering for me...
> 
> RN: 0.54.4
> react-native-swiper: 1.5.13

```
const window = Dimensions.get('window');
  
<------->

 <Swiper ref='mySwiper' onIndexChanged={this.onSwipe}
            loop={false}
            showsPagination={false}
            scrollEnabled={false}
            height={window.height}
            width={window.width}
            containerStyle={{ alignSelf: 'stretch' }}
          >
```
Setting width and height for the swiper fixes the issues on Android. 🚀
